src/Renderer/Shader/PointsVS.glsl

```diff
@@ -83,20 +99,61 @@ void main() {
     } else {
         vColor.a = opacity;
         if (applyOpacityClassication || mode == PNTS_MODE_CLASSIFICATION) {
-            vec2 uv = vec2(classification, 0.5);
-            vColor = texture2D(classificationLUT, uv);
+            vec2 uv = vec2(classification/255., 0.5);
+            vColor = texture2D(classificationTexture, uv);
             vColor.a *= opacity;
         }
 
-        if (mode == PNTS_MODE_INTENSITY) {
-            // adapt the grayscale knowing the range
-            float i = (intensity - intensityRange.x) / (intensityRange.y - intensityRange.x);
-            vColor.rgb = vec3(i, i, i);
-        } else if (mode == PNTS_MODE_NORMAL) {
+        if (mode == PNTS_MODE_NORMAL) {
             vColor.rgb = abs(normal);
         } else if (mode == PNTS_MODE_COLOR) {
             // default to color mode
             vColor.rgb = mix(color, overlayColor.rgb, overlayColor.a);
+        } else if (mode == PNTS_MODE_RETURN_NUMBER) {
+            vec2 uv = vec2(returnNumber/255., 0.5);
+            vColor = texture2D(otherTexture, uv);
+        } else if (mode == PNTS_MODE_RETURN_TYPE) {
+            float returnType;
+            if (returnNumber > numberOfReturns) {
+                returnType = 4.;
+            } else if (returnNumber == 1.) {
+                if (numberOfReturns == 1.) {
+                    // single
+                    returnType = 0.;
+                } else {
+                    // first
+                    returnType = 1.;
+                }
+            } else {
+                if (returnNumber == numberOfReturns) {
+                    // last
+                    returnType = 3.;
+                } else {
+                    // intermediate
+                    returnType = 2.;
+                }
+            }
+            vec2 uv = vec2(returnType/255., 0.5);
+            vColor = texture2D(otherTexture, uv);
+        } else if (mode == PNTS_MODE_RETURN_COUNT) {
+            vec2 uv = vec2(numberOfReturns/255., 0.5);
+            vColor = texture2D(otherTexture, uv);
+        } else if (mode == PNTS_MODE_PTID) {
+            float nb_class = 8.;
```

Comment:
Since this is a constant value use `#define NB_CLASS 8` at top-level to inline this value.